runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -6643,6 +6645,12 @@ static void genInitObjectHeader(TR::Node             *node,
       }
    else
       {
+      // For dynamic array clone, load the array class from the component class and store into clzReg
+      if (dynamicArrayClone)
+         {
+         clzReg = tempReg;
+         generateRegMemInstruction(LRegMem(), node, clzReg, generateX86MemoryReference(classReg, offsetof(J9Class, arrayClass), cg), cg);
+         }	      
```

Comment:
As it stands, this will generate the load twice if dynamic array clone occurs in "old" AOT. Could it be combined with the path that generates the corresponding load above? e.g.
```
if (classReg != NULL && node->getOpCodeValue() == TR::anewarray)
   {
   // Variable component type, due either to dynamicArrayClone or (non-SVM) AOT.
   // Load the array type from it at runtime.
   clzReg = tempReg;
   generateRegMemInstruction(LRegMem(), node, clzReg, generateX86MemoryReference(classReg, offsetof(J9Class, arrayClass), cg), cg);
   }
else if (cg->needClassAndMethodPointerRelocations() && !comp->getOption(TR_UseSymbolValidationManager))
   {
   // ... [check for TR::newarray and TR::New, since TR::anewarray has been ruled out] ...
   }
```